### DISCUSSION TITLE
Enhance tests with `patrol`

### DISCUSSION THREAD

**Discussion by josxha** [josxha/flutter-maplibre#165] [Ideas]
patrol is a UI testing framework for flutter that claims to give control over device permissions.

> Unlock [native platform features](https://patrol.leancode.co/documentation/native/overview) right within your Flutter tests. With Patrol, you can:
> - Interact with permission dialogs, notifications, and WebViews.
> - Modify device settings, toggle Wi-Fi, and more.
> - Achieve all this effortlessly using plain Dart code.

- https://pub.dev/packages/patrol
- https://patrol.leancode.co/
- https://www.youtube.com/watch?v=fidNg4ZzUKA

**Comment by gabbopalma** [comment]
I just saw this patrol live on yt, the package looks very professional and useful
